coretex/entities/dataset/network_dataset.py

```diff
@@ -132,25 +132,38 @@ def _keyDescriptors(cls) -> Dict[str, KeyDescriptor]:
 
         return descriptors
 
-    # NetworkObject overrides
-
     @classmethod
-    def _endpoint(cls) -> str:
-        return "dataset"
+    def decode(cls, encodedObject: Dict[str, Any]) -> Self:
+        obj = super().decode(encodedObject)
 
-    @classmethod
-    def fetchById(cls, objectId: int, **kwargs: Any) -> Self:
-        if "include_sessions" not in kwargs:
-            kwargs["include_sessions"] = 1
+        page = 1
+        hasMorePages = True
+        samples: List[SampleType] = []
 
-        return super().fetchById(objectId, **kwargs)
+        while hasMorePages:
+            params = {
+                "dataset_id": obj.id,
+                "page": page
+            }
+            response = networkManager.get(f"{cls._endpoint()}/sessions", params)
```

Comment:
I tried sending this request through postman, but it seems that pagination is not working this way on backend as it was handled in this code, can you verify this with the backend team?

Reply:
@VukManojlovic did you verify this?

Reply:
I went over this with the backend team. It should be good now.

Reply:
When I talked with Jovica it seems that we either need to wait for the endpoint to be updated or manually send "do_pagination" parameter. I don't see us sending that parameter when sending the request. Can you check again?

Reply:
We tested it successfully